tests/test_pipelines_common.py

```diff
@@ -100,11 +105,14 @@ def get_tiny_tokenizer_from_checkpoint(checkpoint):
     return tokenizer
 
 
-def get_tiny_feature_extractor_from_checkpoint(checkpoint, tiny_config):
+def get_tiny_feature_extractor_from_checkpoint(checkpoint, tiny_config, feature_extractor_class):
     try:
         feature_extractor = AutoFeatureExtractor.from_pretrained(checkpoint)
     except Exception:
-        feature_extractor = None
+        try:
+            feature_extractor = feature_extractor_class()
+        except Exception:
```

Comment:
Can this be more defined than just `Exception` ?

Reply:
Well, I have no idea what could crash on abritraty initiatialization.

If we want to keep the magic being very tolerant here seems OK to me.
The code could fail because `feature_extractor_class` requires some argument to be defined, but also any exception that could be raised during `__init__` so I don't see a way to be exhaustive here.

Reply:
What's the reason this `feature_extractor_class` argument is added?

Reply:
Because we need to initialize the feature extractor.

It does not work with `from_pretrained` above because the `checkpoint == "deepmind/language-perceiver"` (it retrieves from `_CHECKPOINT_FOR_DOC`.

But we do have the class, and feature_extractors usually don't carry a lot of information (like `vocab.txt` with tokenizers I mean) so trying to instantiate with default arguments make sense to me. Otherwise we would need another indirection to make `from_pretrained` work here.